ms2deepscore/data_generators.py

```diff
@@ -138,35 +158,31 @@ def __data_augmentation(self, document_dict):
                 idx = idx[indices]
                 values = values[indices]
         if self.augment_intensity:
+            # TODO: Factor out function with documentation + example?
             values = (1 - self.augment_intensity * 2 * (np.random.random(values.shape) - 0.5)) * values
         return idx, values
 
-
-    def __data_generation(self, list_IDs_temp):
+    def __data_generation(self, batch_ids):
         """Generates data containing batch_size samples"""
         # Initialization
         X = [np.zeros((self.batch_size, self.dim)) for i in range(2)]
         y = np.zeros((self.batch_size,))
 
         # Generate data
-        for i, IDs in enumerate(list_IDs_temp):
-            # Select spectrum for respective inchikey
-            inchikey_1 = self.inchikey_mapping.loc[IDs[0]]["inchikey"]
-            inchikey_2 = self.inchikey_mapping.loc[IDs[1]]["inchikey"]
-            ID1 = np.random.choice(np.where(self.inchikeys_array == inchikey_1)[0])
-            ID2 = np.random.choice(np.where(self.inchikeys_array == inchikey_2)[0])
-
-            ref_idx, ref_values = self.__data_augmentation(self.spectrums_binned_dicts[ID1])
-
-            # Add scaled weight to right bins
-            X[0][i, ref_idx] = ref_values ** self.peak_scaling
-            query_idx, query_values = self.__data_augmentation(self.spectrums_binned_dicts[ID2])
-
-            # Add scaled weight to right bins
-            X[1][i, query_idx] = query_values ** self.peak_scaling
-            y[i] = self.score_array[IDs[0], IDs[1]]
-            if np.isnan(y[i]):
-                y[i] = np.random.random(1)
+        for i_batch, pair in enumerate(batch_ids):
+            for i_pair, id in enumerate(pair):
+                inchikey = self.inchikey_mapping.loc[id]["inchikey"]
+                spectrum_id = np.random.choice(np.where(self.inchikeys_array == inchikey)[0])
```

Comment:
I am not sure if I have the names of ids correct now. There seem to be multiple levels in the hierarchy: training_sample_id -> inchikey -> spectrum_id . Is that correct? I think it would be nice to use these (or better names throughout)

Reply:
changed it in PR #8